### PR TITLE
Add gulp/grunt interfaces for @types/google-closure-compiler

### PR DESCRIPTION
**Pull request by dmchurch**
The google-closure-compiler definition files are missing the bundled gulp/grunt plugins, this adds them. The only thing that might be unusual about this PR is that I've added the grunt type definitions as a `devDependency` rather than a `dependency`, since I wrote the definitions in a way that doesn't directly reference the grunt types. Standard checklist follows:

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/google/closure-compiler-npm/blob/master/packages/google-closure-compiler/docs/gulp.md and https://github.com/google/closure-compiler-npm/blob/master/packages/google-closure-compiler/docs/grunt.md
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Review by sheetalkamat — DISMISSED** [review]
address feedback

File: types/google-closure-compiler/package.json [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
     },
     "devDependencies": {
+        "@types/grunt": "*",
         "@types/google-closure-compiler": "workspace:."
     },
     "owners": [
         {
-            "name": "Evan Martin",
-            "url": "http://neugierig.org"
+            "name": "Evan Martin (http://neugierig.org)",
```
[the remark this answers is not in the document]
Comment (dmchurch) [reply types/google-closure-compiler/package.json:17]: Ah! I didn't think of that, that makes sense. I'll make that change now!

[the remark this answers is not in the document]
Comment (dmchurch) [reply types/google-closure-compiler/package.json]: I did consider that at some length, and my reasoning was that

a) they added themself when they submitted the type definition,
b) using a personal URL that links to their github,
c) at a time when adding a personal URL was a reasonable way to identify yourself in the DT repo,
d) the fact that they haven't _been_ pinged does not mean they didn't _sign up_ to be pinged, and
e) they are perfectly welcome to remove their name from the repo, and making sure they get pinged is a good way to give them that option

**Review by jakebailey — APPROVED** [review]

**Comment by dmchurch** [comment]
Ready to merge

**Pull request merged**